.github/workflows/ci.yml

```diff
@@ -21,15 +21,29 @@ jobs:
       - name: Code Checkout
         uses: actions/checkout@v2
 
-      - name: Prepare environment
+      - name: Toolchain install
         run: make arm_sdk_install
 
-      - name: Execute Build
+      - name: Extra package install
+        run: sudo apt-get install -y libblocksruntime-dev
+
+      - name: Run sanity checks
+        run: make EXTRA_FLAGS=-Werror checks
+
+      - name: Run all unit tests
+        run: make EXTRA_FLAGS=-Werror test-all
+
+      - name: Execute Build (Without Commit Hash)
+        if: inputs.release_build == true
         run: make all
 
+      - name: Execute Build (With Commit Hash)
+        if: inputs.release_build == false
+        run: make all_rev
+
       - name: Publish build artifacts
         uses: actions/upload-artifact@v3
         with:
           name: Assets
           path: ./obj/*.hex
-          retention-days: 7
+          retention-days: 14
```

Comment:
Seeing that we can have PRs pending to be reviewed/merged during a lot of time, maybe is better to increment this? We have a maximum of 90 days available.